src/main/java/wolox/bootstrap/controllers/RoleController.java

```diff
@@ -44,27 +42,27 @@
     private MessageSource messageSource;
 
     @PostMapping
-    public ResponseEntity<Role> create(@RequestBody RoleDto roleDto) throws RoleNotFoundException {
+    public ResponseEntity<Role> create(@RequestBody RoleDto roleDto, UriComponentsBuilder uriComponentsBuilder)
+        throws RoleNotFoundException {
         Preconditions.checkArgument(!roleRepository.findByName(roleDto.getName()).isPresent(),
             messageSource.getMessage(Constants.MSG_CODE_EXISTING_ROLE, null, LocaleContextHolder
                 .getLocale()));
         final Role role = new Role(roleDto.getName());
         log.info("Received role:" + role);
         log.info("Saving role ");
         roleRepository.save(role);
-        final URI uri = ControllerLinkBuilder
-                .linkTo(ControllerLinkBuilder.methodOn(RoleController.class).findById(role.getId())).toUri();
-        return ResponseEntity.created(uri).body(role);
+        final UriComponents uriComponents = uriComponentsBuilder.path("/api/roles/{id}").buildAndExpand(role.getId());
```

Comment:
I'm not sure I like the fact of having to write the URL here, why do you think this is better? Just so I understand.

Reply:
The problem here is that the ControllerLinkBuilder is deprecated in this Spring Boot version, so I had to change it to what they recommend (Which is the `UriComponents`). Anyway, I just did another commit to avoid this hardcoded String 👼